daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala

```diff
@@ -333,4 +337,18 @@ private[data] final class IdStringImpl extends IdString {
   override val ContractIdString: StringModule[ContractIdString] =
     new MatchingStringModule("Daml-LF Contract ID", """#[\w._:\-#/ ]{0,254}""")
 
+  /** Identifiers for participant node users consist of ASCII digits and lower-case alphabetic characters,
+   *  hyphens, underscores, and dots, and satisfy the following  rules:
+   *  1. The characters `-._` never follow each other.
+   *  2. The characters `-._` neither occur at the start nor at the end of the user name.
+   *  Thus 'john.doe1' is a valid user-name, while 'john..doe1', 'john.-doe1', and '-john.doe' are not.
+   */
+  override type UserId = String
+  override val UserId: StringModule[UserId] =
+    new MatchingStringModule(
+      "User ID",
+      """[\p{Lower}\d]([\p{Lower}\d]|[-._][\p{Lower}\d]){0,62}"""
```

Comment:
why 62?

Reply:
Limit to at most 63 characters, which is a shorter variant of how we limit ledger strings to 255 characters.